lib/improver/tests/orographic_enhancement/test_OrographicEnhancement.py

```diff
@@ -0,0 +1,839 @@
+    """Test the _calculate_svp method"""
+
+    def test_basic(self):
+        """Test output is a cube"""
+        self.plugin._calculate_svp(method='IMPROVER')
+        self.assertIsInstance(self.plugin.svp, iris.cube.Cube)
+
+    def test_values(self):
+        """Test output values from the IMPROVER method"""
+        expected_data = np.array([
+            [813.645300, 878.022313, 848.258948],
+            [903.222415, 890.543221, 922.188935],
+            [922.147421, 928.393655, 967.875963]])
+        self.plugin._calculate_svp(method='IMPROVER')
+        self.assertArrayAlmostEqual(self.plugin.svp.data, expected_data)
+
+    def test_values_steps(self):
+        """Test output values from the STEPS method"""
+        expected_data = np.array([
+            [810.056593, 874.936670, 844.897647],
+            [899.643230, 887.213984, 918.575494],
+            [918.575494, 924.963966, 964.125249]])
+        self.plugin._calculate_svp(method='STEPS')
+        self.assertArrayAlmostEqual(self.plugin.svp.data, expected_data)
+
+
+class Test__generate_mask(IrisTest):
+    """Test the _generate_mask method"""
+
+    def setUp(self):
+        """Set up and populate a plugin instance"""
+        self.plugin = OrographicEnhancement()
+        x_coord = DimCoord(np.arange(5), 'projection_x_coordinate',
+                           units='km')
+        y_coord = DimCoord(np.arange(5), 'projection_y_coordinate',
+                           units='km')
+
+        # this is neighbourhood-processed as part of mask generation
+        topography_data = np.array([[0., 10., 20., 50., 100.],
+                                    [10., 20., 50., 100., 200.],
+                                    [25., 60., 80., 160., 220.],
+                                    [50., 80., 100., 200., 250.],
+                                    [50., 80., 100., 200., 250.]])
+        self.plugin.topography = iris.cube.Cube(
+            topography_data, long_name="topography", units="m",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+
+        humidity_data = np.full((5, 5), 0.9)
+        humidity_data[1, 3] = 0.5
+        self.plugin.humidity = iris.cube.Cube(
+            humidity_data, long_name="relhumidity", units="1",
+            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+
+        self.plugin.vgradz = np.full((5, 5), 0.01)
+        self.plugin.vgradz[3:, :] = 0.
+
+    def test_basic(self):
+        """Test output is array"""
+        result = self.plugin._generate_mask()
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_values(self):
+        """Test output mask is correct"""
+        expected_output = np.array([[True, True, False, False, False],
+                                    [False, False, False, True, False],
+                                    [False, False, False, False, False],
+                                    [True, True, True, True, True],
+                                    [True, True, True, True, True]])
```

Comment:
I like unit-tests to be really easy to understand. Could you define your expected output in the same terms you define the inputs? So, your humidity test point is [1, 3] => expected_output[1, 3] = True.
What is currently here works, but requres too much mental agility.